### PR TITLE
SUL23-489: Places to study table.

### PR DESCRIPTION
**Pull request by jenbreese**
# READY FOR REVIEW

# Summary
- This is a table view of the Places to Study.
- It has filtering at the top.

# Review By (Date)
- 6/24

# Criticality
- High

# Urgency
- High

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Rebuild Cache and import config drush cr ; drush ci
3. Navigate to a Basic page and add a list of Places to Study. Select the Places to Study Table view.
4. Verify it looks like the design in all the responsive views. There is a small, med and large layout.
5. Verify the filters are working as expected.

### Site Configuration Sync

- Is there a config:export in this PR that changes the config sync directory?

## Front End Validation
- [ ] Design is approved by @ user?
- [ ] HTML validation: Is the markup using the appropriate semantic tags and [passes validation](https://validator.w3.org/nu/)? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Cross-browser testing: Has been performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Automated accessibility: Scans performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Manual accessibility: Manually tested? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?

### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
[- SUL23-442](https://stanfordits.atlassian.net/browse/SUL23-442)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

### CODE REVIEW

File: src/components/views/sul-study-place/study-place-today-hours-table.tsx [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+}
+
+const StudyPlaceHoursComponent = ({hoursId}: { hoursId: string }) => {
+  const hours = useTodayLibraryHours(hoursId);
+  if (!hours) {
+    return null;
+  }
+
+  const {closedAllDay, isOpen, openingTime, closingTime, afterClose} = hours;
+  const openUntil = "Open until ";
```
Comment (jenbreese) [inline src/components/views/sul-study-place/study-place-today-hours-table.tsx:22]: @pookmish I'd like to put a class on this Open but we can't have html in the variable and string them together like the olden days.  What is a better way to think about this challenge?

Comment (pookmish) [reply src/components/views/sul-study-place/study-place-today-hours-table.tsx:22]: put the variable into the markup:
`<span className="foobar">{openUntil}</span><span> More Stuff</span>`

Comment (pookmish) [reply src/components/views/sul-study-place/study-place-today-hours-table.tsx:22]: You should always use the tags and markup in the component, not in variables.

**Review by pookmish — APPROVED** [review]
I want to get some a11y eyes on the hover/checked/focus state of the radio buttons. but we can do that next.

**Pull request merged**